pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -225,6 +225,7 @@ func readConfig(config io.Reader) (VSphereConfig, error) {
 }
 
 func init() {
+	registerMetrics()
```

Comment:
An error is returned only when collector being registered is incompatible or inconsistent with already registered metrics. we have fixed statically defined compatible collector metrics - vsphereApiErrorMetric and vsphereApiMetric, so should never panic.

Other providers GCE, AWS also uses prometheus.MustRegister.

We can read and discuss in person. 
https://github.com/prometheus/client_golang/blob/master/prometheus/registry.go#L87